bids/analysis/transformations/compute.py

```diff
@@ -52,8 +48,14 @@ def _transform(self, var, model='spm', derivative=False, dispersion=False,
         convolved = hrf.compute_regressor(vals, model, onsets,
                                           fir_delays=fir_delays, min_onset=0)
 
-        return DenseRunVariable(name=var.name, values=convolved[0], run_info=var.run_info,
-                                source=var.source, sampling_rate=var.sampling_rate)
+        if isinstance(var, SparseRunVariable):
+            return SparseRunVariable(
+                name=var.name, values=convolved[0], onset=onsets,
```

Comment:
Unrelated to the pr, but unchecked assumptions started as `[0]` is a recipe for a trouble or a cryptic error. I believe we addressed some of such, but it would be nice if new code with checks/proper errors or at least assert statements

Reply:
I blame someone else!  Maybe if it returns a dictionary?